flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java

```diff
@@ -0,0 +1,229 @@
+            }
+
+            if (spec.getJobManager() != null) {
+                if (spec.getJobManager().getResource() != null) {
+                    effectiveConfig.setString(
+                            JobManagerOptions.TOTAL_PROCESS_MEMORY.key(),
+                            spec.getJobManager().getResource().getMemory());
+                    effectiveConfig.set(
+                            KubernetesConfigOptions.JOB_MANAGER_CPU,
+                            spec.getJobManager().getResource().getCpu());
+                }
+
+                if (spec.getJobManager().getPodTemplate() != null) {
+                    effectiveConfig.set(
+                            KubernetesConfigOptions.JOB_MANAGER_POD_TEMPLATE,
+                            createTempFile(
+                                    mergePodTemplates(
+                                            spec.getPodTemplate(),
+                                            spec.getJobManager().getPodTemplate())));
+                }
+            }
+
+            if (spec.getTaskManager() != null) {
+                if (spec.getTaskManager().getTaskSlots() > 0) {
+                    effectiveConfig.set(
+                            TaskManagerOptions.NUM_TASK_SLOTS,
+                            spec.getTaskManager().getTaskSlots());
+                }
+
+                if (spec.getTaskManager().getResource() != null) {
+                    effectiveConfig.setString(
+                            TaskManagerOptions.TOTAL_PROCESS_MEMORY.key(),
+                            spec.getTaskManager().getResource().getMemory());
+                    effectiveConfig.set(
+                            KubernetesConfigOptions.TASK_MANAGER_CPU,
+                            spec.getTaskManager().getResource().getCpu());
+                }
+
+                if (spec.getTaskManager().getPodTemplate() != null) {
+                    effectiveConfig.set(
+                            KubernetesConfigOptions.TASK_MANAGER_POD_TEMPLATE,
+                            createTempFile(
+                                    mergePodTemplates(
+                                            spec.getPodTemplate(),
+                                            spec.getTaskManager().getPodTemplate())));
+                }
+            }
+
+            if (spec.getJob() != null) {
+                final URI uri = new URI(spec.getJob().getJarURI());
+                effectiveConfig.set(
+                        PipelineOptions.JARS, Collections.singletonList(uri.toString()));
+
+                if (spec.getJob().getParallelism() > 0) {
+                    effectiveConfig.set(
+                            CoreOptions.DEFAULT_PARALLELISM, spec.getJob().getParallelism());
+                }
+            }
+
+            return effectiveConfig;
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to load configuration", e);
+        }
+    }
```

Comment:
This method is very long, suggest splitting it out into smaller methods

Reply:
I completely agree, I would not jump on this straight away but will add a ticket for this so we don't forget.

Reply:
https://issues.apache.org/jira/browse/FLINK-26163